### ISSUE TITLE
100% class error but loss is 0

### ISSUE THREAD

**Issue by DableUTeeF** [IDEACVR/DINO#106]
I trained ResNet 50 model on colab with
```
python main.py \
	--output_dir /content/drive/MyDrive/Chula_Fluke/checkpoints_fixed_old_box/DINO_R50_4scale -c config/DINO/DINO_4scale.py --coco_path /content/Chula-ParasiteEgg-11 \
    --pretrain_model_path r50_4scale.pth \
	--num_workers 0 \
	--options dn_scalar=100 embed_init_tgt=TRUE \
	dn_label_coef=1.0 dn_bbox_coef=1.0 use_ema=False \
	dn_box_noise_scale=1.0 backbone_dir=. batch_size=2 \
    num_classes=12 
```
I used a pretrained mscoco weight but slightly modified the `class_embed` layer so that it work with `--pretrained_model_path` using 
```
import collections
sd = torch.load('checkpoint0033_4scale.pth')
new_model = collections.OrderedDict()
for k, v in sd['model'].items():
    if 'class_embed' not in k:
        new_model[k] = v
```
I noticed that if I change the `class_embed` weight instead of completely removed it the ce_loss of the first few iterations is no longer 0 but still gradually dropped to 0 afterward.

Here's the last line of log before I terminated the training.
```
Epoch: [5]  [ 700/5628]  eta: 3:36:58  lr: 0.000100  class_error: 100.00  loss: 0.0000 (0.0000)  loss_bbox_dn: 0.0000 (0.0000)  loss_giou_dn: 0.0000 (0.0000)  loss_ce_dn: 0.0000 (0.0000)  loss_ce: 0.0000 (0.0000)  loss_bbox: 0.0000 (0.0000)  loss_giou: 0.0000 (0.0000)  loss_ce_0: 0.0000 (0.0000)  loss_bbox_0: 0.0000 (0.0000)  loss_giou_0: 0.0000 (0.0000)  loss_bbox_dn_0: 0.0000 (0.0000)  loss_giou_dn_0: 0.0000 (0.0000)  loss_ce_dn_0: 0.0000 (0.0000)  loss_ce_1: 0.0000 (0.0000)  loss_bbox_1: 0.0000 (0.0000)  loss_giou_1: 0.0000 (0.0000)  loss_bbox_dn_1: 0.0000 (0.0000)  loss_giou_dn_1: 0.0000 (0.0000)  loss_ce_dn_1: 0.0000 (0.0000)  loss_ce_2: 0.0000 (0.0000)  loss_bbox_2: 0.0000 (0.0000)  loss_giou_2: 0.0000 (0.0000)  loss_bbox_dn_2: 0.0000 (0.0000)  loss_giou_dn_2: 0.0000 (0.0000)  loss_ce_dn_2: 0.0000 (0.0000)  loss_ce_3: 0.0000 (0.0000)  loss_bbox_3: 0.0000 (0.0000)  loss_giou_3: 0.0000 (0.0000)  loss_bbox_dn_3: 0.0000 (0.0000)  loss_giou_dn_3: 0.0000 (0.0000)  loss_ce_dn_3: 0.0000 (0.0000)  loss_ce_4: 0.0000 (0.0000)  loss_bbox_4: 0.0000 (0.0000)  loss_giou_4: 0.0000 (0.0000)  loss_bbox_dn_4: 0.0000 (0.0000)  loss_giou_dn_4: 0.0000 (0.0000)  loss_ce_dn_4: 0.0000 (0.0000)  loss_ce_interm: 0.0000 (0.0000)  loss_bbox_interm: 0.0000 (0.0000)  loss_giou_interm: 0.0000 (0.0000)  loss_bbox_dn_unscaled: 0.0000 (0.0000)  loss_giou_dn_unscaled: 0.0000 (0.0000)  loss_ce_dn_unscaled: 0.0000 (0.0000)  loss_xy_dn_unscaled: 0.0000 (0.0000)  loss_hw_dn_unscaled: 0.0000 (0.0000)  cardinality_error_dn_unscaled: 0.0000 (0.0000)  loss_ce_unscaled: 0.0000 (0.0000)  class_error_unscaled: 100.0000 (100.0000)  loss_bbox_unscaled: 0.0000 (0.0000)  loss_giou_unscaled: 0.0000 (0.0000)  loss_xy_unscaled: 0.0000 (0.0000)  loss_hw_unscaled: 0.0000 (0.0000)  cardinality_error_unscaled: 900.0000 (900.0000)  loss_ce_0_unscaled: 0.0000 (0.0000)  loss_bbox_0_unscaled: 0.0000 (0.0000)  loss_giou_0_unscaled: 0.0000 (0.0000)  loss_xy_0_unscaled: 0.0000 (0.0000)  loss_hw_0_unscaled: 0.0000 (0.0000)  cardinality_error_0_unscaled: 900.0000 (900.0000)  loss_bbox_dn_0_unscaled: 0.0000 (0.0000)  loss_giou_dn_0_unscaled: 0.0000 (0.0000)  loss_ce_dn_0_unscaled: 0.0000 (0.0000)  loss_xy_dn_0_unscaled: 0.0000 (0.0000)  loss_hw_dn_0_unscaled: 0.0000 (0.0000)  cardinality_error_dn_0_unscaled: 0.0000 (0.0000)  loss_ce_1_unscaled: 0.0000 (0.0000)  loss_bbox_1_unscaled: 0.0000 (0.0000)  loss_giou_1_unscaled: 0.0000 (0.0000)  loss_xy_1_unscaled: 0.0000 (0.0000)  loss_hw_1_unscaled: 0.0000 (0.0000)  cardinality_error_1_unscaled: 900.0000 (900.0000)  loss_bbox_dn_1_unscaled: 0.0000 (0.0000)  loss_giou_dn_1_unscaled: 0.0000 (0.0000)  loss_ce_dn_1_unscaled: 0.0000 (0.0000)  loss_xy_dn_1_unscaled: 0.0000 (0.0000)  loss_hw_dn_1_unscaled: 0.0000 (0.0000)  cardinality_error_dn_1_unscaled: 0.0000 (0.0000)  loss_ce_2_unscaled: 0.0000 (0.0000)  loss_bbox_2_unscaled: 0.0000 (0.0000)  loss_giou_2_unscaled: 0.0000 (0.0000)  loss_xy_2_unscaled: 0.0000 (0.0000)  loss_hw_2_unscaled: 0.0000 (0.0000)  cardinality_error_2_unscaled: 900.0000 (899.9979)  loss_bbox_dn_2_unscaled: 0.0000 (0.0000)  loss_giou_dn_2_unscaled: 0.0000 (0.0000)  loss_ce_dn_2_unscaled: 0.0000 (0.0000)  loss_xy_dn_2_unscaled: 0.0000 (0.0000)  loss_hw_dn_2_unscaled: 0.0000 (0.0000)  cardinality_error_dn_2_unscaled: 0.0000 (0.0000)  loss_ce_3_unscaled: 0.0000 (0.0000)  loss_bbox_3_unscaled: 0.0000 (0.0000)  loss_giou_3_unscaled: 0.0000 (0.0000)  loss_xy_3_unscaled: 0.0000 (0.0000)  loss_hw_3_unscaled: 0.0000 (0.0000)  cardinality_error_3_unscaled: 900.0000 (899.9800)  loss_bbox_dn_3_unscaled: 0.0000 (0.0000)  loss_giou_dn_3_unscaled: 0.0000 (0.0000)  loss_ce_dn_3_unscaled: 0.0000 (0.0000)  loss_xy_dn_3_unscaled: 0.0000 (0.0000)  loss_hw_dn_3_unscaled: 0.0000 (0.0000)  cardinality_error_dn_3_unscaled: 0.0000 (0.0000)  loss_ce_4_unscaled: 0.0000 (0.0000)  loss_bbox_4_unscaled: 0.0000 (0.0000)  loss_giou_4_unscaled: 0.0000 (0.0000)  loss_xy_4_unscaled: 0.0000 (0.0000)  loss_hw_4_unscaled: 0.0000 (0.0000)  cardinality_error_4_unscaled: 900.0000 (900.0000)  loss_bbox_dn_4_unscaled: 0.0000 (0.0000)  loss_giou_dn_4_unscaled: 0.0000 (0.0000)  loss_ce_dn_4_unscaled: 0.0000 (0.0000)  loss_xy_dn_4_unscaled: 0.0000 (0.0000)  loss_hw_dn_4_unscaled: 0.0000 (0.0000)  cardinality_error_dn_4_unscaled: 0.0000 (0.0000)  loss_ce_interm_unscaled: 0.0000 (0.0000)  loss_bbox_interm_unscaled: 0.0000 (0.0000)  loss_giou_interm_unscaled: 0.0000 (0.0000)  loss_xy_interm_unscaled: 0.0000 (0.0000)  loss_hw_interm_unscaled: 0.0000 (0.0000)  cardinality_error_interm_unscaled: 900.0000 (899.9308)  time: 2.7692  data: 0.2870  max mem: 12619
```

**Comment by EmpireShadow** [comment]
how can you deal with this problem

**Comment by EmpireShadow** [comment]
how can you deal with this problem

**Comment by DableUTeeF** [comment]
```
import collections
sd = torch.load('checkpoint0033_4scale.pth')
new_model = collections.OrderedDict()
for k, v in sd['model'].items():
    if 'class_embed' not in k:
        new_model[k] = v
    else:
        new_model[k] = v[10:17]
```
Then 
```
torch.save({
    'optimizer': sd['optimizer'], 
    'lr_scheduler': sd['lr_scheduler'], 
    'epoch': sd['epoch'], 
    'args': sd['args'],
    'model': new_model
    }, '4scale_swin.pth'
)
```
And use that `4scale_swin.pth` for `--pretrain_model_path`.

**Comment by DableUTeeF** [comment]
The `v[10:17]` is in case the new data has 7 classes. No particular reason for why it is 10:17 and since it change the position it's like randomizing again anyway. But you need to have `class_embed` with the correct size in the weight. Doesn't really matter what it previously trained for.

**Comment by EmpireShadow** [comment]
thanks bro , im trying this method on my code
